backend/tests/modules/user/allocations/test_pending_allocations.py

```diff
@@ -70,6 +74,7 @@ def service(
     return PendingUserAllocations(
```

Comment:
Please add two tests:
- to check if the uq is saved
- to check if uq score is not updated with a second allocation within the same epoch